test/extended/testdata/imagestream-jenkins-slave-pods.yaml

```diff
@@ -0,0 +1,22 @@
+apiVersion: v1
+kind: ImageStream
+metadata:
+  annotations:
+    slave-label: asdf
+  labels:
+    role: jenkins-slave
+  name: asdf
+spec:
+  tags:
+  - annotations:
+      role: jenkins-slave
+    from:
+      kind: DockerImage
+      name: docker.io/openshift/jenkins-slave-maven-centos7:latest
+    name: base
+  - annotations:
+      role: jenkins-slave
+    from:
+      kind: ImageStreamTag
+      name: base
+    name: latest
```

Comment:
i'm confused by this file... you've got the imagestream labeled but also two ISTs annotated, at least one of which you don't appear to be explicitly using in the tests.

What i would have expected is two tests:

1) define an IS that has no explicit tags (just define a repository on the IS itself and let it import tags) - has the role=jenkins-slave + slave-label annotation.

2) define an IS that does NOT have the role itself, but does define an IST that has the jenkins-slave annotation.